### ISSUE TITLE
amountLimiting option not apply if no limit filter GraphQL

### ISSUE THREAD

**Issue by conghai** [strapi/strapi#2524]
`amountLimit` in `config/settings.json` of plugin `graphql` only apply when `limit` param has been set:
```
// ex: amountLimit: 50
query {
   posts(limit: 100) {
      _id
      title
   }
}
```
=> this results have 50 record (it's right)

**BUT**
```
// amountLimit: 50 & without limit param
query {
   posts {
      _id
      title
   }
}
```
=> this results have all record in your database (not limit apply)

-----
**Solution to Fix**
```
/* 
* file: services/Query.js
* line: 34 - 42
* line 37: if(!params.limit) -> set  params.limit = strapi.plugins.graphql.config.amountLimit;
*/
amountLimiting: params => {
    if (params.limit && params.limit < 0) {
      params.limit = 0;
    } else if (!params.limit || params.limit > strapi.plugins.graphql.config.amountLimit) {
      params.limit = strapi.plugins.graphql.config.amountLimit;
    }

    return params;
  }
```

**Comment by derrickmehaffy** [comment]
good catch, I wrote the patch for that half asleep lol. If you want to submit a PR to get credit for the code by all means go for it 😄

**Comment by lauriejim** [comment]
Hello @conghai , thank you for reporting this issue. We are currently on another feature and bug fix. Since time is lacking on our side, feel free investigate and submit a PR, we’ll appreciate your contribution on this issue! 

You look have the solution, I will appreciate if you submit a PR to fix it.
Thank you!

Check out the contributing guide to get started: https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md

**Comment by lauriejim** [comment]
Hey I see your commit but there is no PR, have you some news about this point ?

**Comment by yurijmikhalevich** [comment]
This was fixed in `v3.0.0-beta.4` and can be closed. Related Issue https://github.com/strapi/strapi/issues/3420. @derrickmehaffy
